SampleGallery/Samples/SDK 14393/Win2DInterop/Win2DInterop.xaml.cs

```diff
@@ -0,0 +1,258 @@
+﻿using Microsoft.Graphics.Canvas.Effects;
+using Microsoft.Graphics.Canvas.UI.Composition;
+using SamplesCommon;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace CompositionSampleGallery
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class Win2DInterop : SamplePage
+    {
+        public Win2DInterop()
+        {
+            this.InitializeComponent();
+        }
+
+        public static string StaticSampleName { get { return "Win2D Interop"; } }
+        public override string SampleName { get { return StaticSampleName; } }
+        public override string SampleDescription { get { return "Drawing a Win2D effect into a CompositionDrawingSurface so it can be used as part of a composition effect graph. A noise bitmap is generated using the Win2D TurbulenceEffectNoise effect and then used as the alpha channel of the content bitmap."; } }
+
+        #region UI Event Handlers
+
+        private void Page_Loaded(object sender, RoutedEventArgs e)
+        {
+            NoiseOctavesSlider.Value = _noiseOctaves;
+            NoiseFrequencySlider.Value = _noiseFrequency.X;
+            if (_noiseType == TurbulenceEffectNoise.FractalSum)
+            {
+                NoiseTypeComboBox.SelectedValue = "Fractal Sum";
+            }
+            else
+            {
+                NoiseTypeComboBox.SelectedValue = "Turbulence";
+            }
+
+            var fireAndForget = SetupEffect();
+        }
+
+        private void Page_Unloaded(object sender, RoutedEventArgs e)
+        {
+            DisposeEffect();
+        }
+
+        private void Placeholder_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            ResizeEffect();
+        }
+
+        private void RebuildButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (!_initialized)
+            {
+                return;
+            }
+
+            var fireAndForget = SetupEffect();
```

Comment:
To enable dynamic updates from directly from the slider/combo updates, consider making SetupEffect() a one time initialization function for page load and create a separate UpdateSurface() method which is called in response to UI updates.